sor/src/main/java/com/bazaarvoice/emodb/sor/core/DefaultDataStore.java

```diff
@@ -1025,4 +997,61 @@ private void decrementDeltaSizes(PendingCompaction pendingCompaction) {
     private String getMetricName(String name) {
         return MetricRegistry.name("bv.emodb.sor", "DefaultDataStore", name);
     }
+
+    private Iterator<RecordUpdate> transformUpdates(Iterator<Update> updatesIter, boolean isFacade, final Set<String> tags) {
+        return Iterators.transform(updatesIter, new Function<Update, RecordUpdate>() {
+            @Override
+            public RecordUpdate apply(Update update) {
+                requireNonNull(update, "update");
+                String tableName = update.getTable();
+                String key = update.getKey();
+                UUID changeId = update.getChangeId();
+                Delta delta = update.getDelta();
+                Audit audit = update.getAudit();
+
+                // Strip intrinsics and "~tags".  Verify the Delta results in a top-level object.
+                delta = SanitizeDeltaVisitor.sanitize(delta);
+
+                Table table = _tableDao.get(tableName);
+
+                if (isFacade && !table.isFacade()) {
+                    // Someone is trying to update a facade, but is inadvertently going to update the primary table in this dc
+                    throw new SecurityException("Access denied. Update intended for a facade, but the table would be updated.");
+                }
+
+                if (table.isFacade() && !isFacade) {
+                    throw new SecurityException("Access denied. Unauthorized attempt to update a facade.");
+                }
+
+                // We'll likely fail to resolve write conflicts if deltas are written into the far past after
+                // compaction may have occurred.
+                if (TimeUUIDs.getTimeMillis(changeId) <= _dataWriterDao.getFullConsistencyTimestamp(table)) {
+                    throw new IllegalArgumentException(
+                            "The 'changeId' UUID is from too far in the past: " + TimeUUIDs.getDate(changeId));
+                }
+
+                return new RecordUpdate(table, key, changeId, delta, audit, tags, update.getConsistency());
+            }
+        });
+    }
+
+    @Override
+    public void updateRefInDatabus(Iterable<Update> updates, Set<String> tags, boolean isFacade) {
+        Iterator<Update> updatesIter = updates.iterator();
+        if (!updatesIter.hasNext()) {
+            return;
+        }
+        Iterator<RecordUpdate> recordUpdates = transformUpdates(updatesIter, isFacade, tags);
+
+        while (recordUpdates.hasNext()) {
+            RecordUpdate update = recordUpdates.next();
+            List<UpdateRef> updateRefs = Lists.newArrayListWithCapacity(Collections.singleton(update).size());
```

Comment:
Why is 'Collections.singleton(update)' required to get size, while size of a Collection.singleton(Set) will be 1?
Refer: https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/master/src/java.base/share/classes/java/util/Collections.java#L4788

Reply:
this piece of code is modified and is no longer in use.